### DISCUSSION TITLE
How to get list results from QVariant

### DISCUSSION THREAD

**Discussion by RuoyanMeng** [MeVisLab/pythonqt#98] [Q&A] [answered]
Hi, I am using pythonQt to embed python code in my C++ code, I create a python function as the example did, but my return value is a list, and I couldn't fetch the list value after calling the function
`context.evalScript("def multiply(a,b):\n  return [a,b];\n");`
`QVariantList args;`
`args << 42 << 47;`
`QVariant result = context.call("multiply", args);` 

Could you give some idea on how to get the result when the return value is a list?

**Comment by usiems** [comment, marked as answer]
```
QVariantList resultList = result.toList();
int a = resultList[0].toInt();
int b = resultList[1].toInt();
```
This is without any type checking, but I guess the important part is the toList() method. Simply have a look at the interface of QVariant.

**Comment by RuoyanMeng** [comment]
Thank you so much! I tried the toList method before, but I didn't notice
that toInt() is needed to extract the corresponding data type value.
